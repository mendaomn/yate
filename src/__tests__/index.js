const yate = require('../index.js')

describe('yate.expect', () => {
  test('toBe: should match strictly equal values', () => {
    expect(() => yate.expect(42).toBe(42)).not.toThrow()
    expect(() => yate.expect("string").toBe("string")).not.toThrow()
    expect(() => yate.expect(true).toBe(true)).not.toThrow()
  });
  
  test('toEqual: should match objects and arrays', () => {
    expect(() => yate.expect({a: 1}).toEqual({a: 1})).not.toThrow()
    expect(() => yate.expect([4, 2]).toEqual([4, 2])).not.toThrow()
  });

  test('should throw when doesn\'t match', () => {
    expect(() => yate.expect(42).toBe(43)).toThrowErrorMatchingSnapshot()
    expect(() => yate.expect("string").toBe("STRING")).toThrowErrorMatchingSnapshot()
    expect(() => yate.expect(true).toBe(false)).toThrowErrorMatchingSnapshot()

    expect(() => yate.expect({a: 1}).toEqual({a: 2})).toThrowErrorMatchingSnapshot()
    expect(() => yate.expect([2, 4]).toEqual([4, 2])).toThrowErrorMatchingSnapshot()
  });
})

describe('yate.test', () => {
  let actualConsoleLog = console.log
  let actualConsoleError = console.error
  
  beforeAll(() => {
    console.log = jest.fn()
    console.error = jest.fn()
  })

  afterAll(() => {
    console.log = actualConsoleLog
    console.error = actualConsoleError
  })

  test('should run the test function', () => {
    const title = 'test title'
    const testFn = jest.fn()

    yate.test(title, testFn)

    expect(testFn).toHaveBeenCalledTimes(1)
    expect(testFn).toHaveBeenCalledWith()
    expect(console.log).toMatchSnapshot()
  });

  test('should catch exceptions thrown by the test function', () => {
    const title = 'test title'
    const testFn = jest.fn(() => { throw new Error('should\'t fire') })

    expect(() => yate.test(title, testFn)).not.toThrow()
    expect(console.error).toMatchSnapshot()
  });

  test('should log a helpful error message', () => {

  })
});

describe('yate.spy', () => {
  test('should track calls and arguments', () => {
    const fn = yate.spy()

    fn('will be lost')
    fn(42)

    expect(fn).toHaveProperty('__mock')
    expect(fn.__mock).toHaveProperty('calls')
    expect(fn.__mock.calls).toBe(2)
    expect(fn.__mock.args).toBe(42)
  });
});