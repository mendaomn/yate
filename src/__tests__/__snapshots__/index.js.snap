// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yate.expect should throw when doesn't match 1`] = `"42 is not equal to 43"`;

exports[`yate.expect should throw when doesn't match 2`] = `"string is not equal to STRING"`;

exports[`yate.expect should throw when doesn't match 3`] = `"true is not equal to false"`;

exports[`yate.expect should throw when doesn't match 4`] = `"{\\"a\\":1} is not equal to {\\"a\\":2}"`;

exports[`yate.expect should throw when doesn't match 5`] = `"[2,4] is not equal to [4,2]"`;

exports[`yate.test should catch exceptions thrown by the test function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "FAIL: test title
    
    Error: should't fire",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`yate.test should run the test function 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "PASS: ",
      "test title",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
